broker/cloud_functions/lsst/ps_to_gcs/main.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
+
+"""This module stores the alert data as an Avro file in Cloud Storage,
+fixing the schema first, if necessary.
+"""
+
+import base64
+import io
+import json
+import os
+import struct
+from typing import Optional
+
+import fastavro
+import pittgoogle
+from confluent_kafka.schema_registry import SchemaRegistryClient
+from google.cloud import functions_v1, logging, storage, pubsub_v1
+from google.cloud.exceptions import PreconditionFailed
+
+
+PROJECT_ID = os.getenv("GCP_PROJECT")
+TESTID = os.getenv("TESTID")
+SURVEY = os.getenv("SURVEY")
+VERSIONTAG = os.getenv("VERSIONTAG")
+
+# connect to the cloud logger and publisher
+logging_client = logging.Client()
+log_name = "ps-to-gcs-cloudfnc"
+logger = logging_client.logger(log_name)
+publisher = pubsub_v1.PublisherClient()
+
+# GCP resources used in this module
+client = storage.Client()
```

Comment:
Pre-defining the buckets and making a `.get_bucket` request for each _instance_ rather than each alert is a temporary solution I discussed with @troyraen. For future schema implementations, this will need to be updated